### PR TITLE
Add tests and refactor parse_blast

### PR DESCRIPTION
**Pull request by stevenhwu**
- Refactor into several small functions
- Add pytest for some of these new functions

### CODE REVIEW

File: src/parse_ncbi_xml.py [hunk 1]
```diff
@@ -1,54 +1,93 @@
...
+
+def parse_alignment_title(alignment_title):
+    alignment_id_full, description = alignment_title.split(" ", 2)[1:3]
+
+    alignment_id = alignment_id_full.split("|")[1]
+    element = REGEX_ELEMENTS.search(description)
+    labels, organism, existence = element.groups()
+
+    enzyme_id, enzyme_code, product_name = parse_product_split(labels)
+    enzyme_id, enzyme_code, product_name = parse_product_regex(labels)
```
Comment (yehzx) [inline src/parse_ncbi_xml.py:62]: Are there any reasons that we need two separate functions to parse the product?
`parse_product_regex` itself seems enough to me.

File: src/parse_ncbi_xml.py [hunk 2]
```diff
@@ -1,54 +1,93 @@
...
+REGEX_PRODUCT = re.compile("^(.+)~{3}(.+)~{3}(.+)$")
+REGEX_GENE = re.compile("GN=(.*) PE=")
+
+
+def parse_product_split(labels):
+    # if labeled: {enzyme_id}~~~{enzyme_code}~~~{product}
+    # if not    : {product}
+    labels_split = labels.replace(",", "").split("~~~")
+    try:
+        enzyme_id, enzyme_code, product_name = labels_split[0:4]
```
Comment (yehzx) [inline src/parse_ncbi_xml.py:31]: labels_split[0:3]?
Though I've checked that labels_split[0:4] won't raise IndexError.

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+
+[options]
+package_dir =
+  = src
+packages = find:
+python_requires = >=3.7
+
+[options.packages.find]
+where = src
```
Comment (yehzx) [inline setup.cfg:12]: I see in setuptools's document we can use pyproject.toml to set up setuptools.
<https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html>
Are we going to merge this to pyproject.toml? Or are there any good reasons that we should keep this file?

File: src/parse_ncbi_xml.py (hunk 2 above)
Comment (stevenhwu) [reply src/parse_ncbi_xml.py]: slicing won't cause errors.
raise is caused by assigning results into 3 variables,

File: src/parse_ncbi_xml.py (hunk 1 above)
Comment (stevenhwu) [reply src/parse_ncbi_xml.py:64]: No, there are exactly the same just two different approaches.

File: setup.cfg (hunk 3 above)
Comment (stevenhwu) [reply setup.cfg:12]: for the package, no, toml is the predicted future.
for testing with pytest, ya, setup files are still needed to test without installing

File: src/parse_ncbi_xml.py (hunk 2 above)
Comment (yehzx) [reply src/parse_ncbi_xml.py]: Now I get it. If it's [0:3], then the potential error (split into more than 3 variables) cannot be detected and caught.
I'll change it back to [0:4].

File: src/parse_ncbi_xml.py (hunk 1 above)
Comment (yehzx) [reply src/parse_ncbi_xml.py:64]: Maybe I'll pick the second one (`parse_product_regex`). 
There're only three cases that will happen: 1) can be split into 3 variables, 2) cannot be split, so only 1, 3) split into 2 or more than 3 variables. (so value errors should be raised)
The second function deals with these better.

**Pull request merged**